mailbox/api/src/main/java/org/apache/james/mailbox/acl/UnionMailboxACLResolver.java

```diff
@@ -240,17 +238,17 @@ public MailboxACL applyGlobalACL(MailboxACL resourceACL, boolean resourceOwnerIs
      * @see org.apache.james.mailbox.store.mail.MailboxACLResolver#hasRight(java.
      *      lang.String, org.apache.james.mailbox.store.mail.MailboxACLResolver.
      *      GroupMembershipResolver,
-     *      org.apache.james.mailbox.MailboxACL.MailboxACLRight,
+     *      org.apache.james.mailbox.MailboxACL.Right,
      *      org.apache.james.mailbox.MailboxACL, java.lang.String)
      */
     @Override
-    public boolean hasRight(String requestUser, GroupMembershipResolver groupMembershipResolver, MailboxACLRight right, MailboxACL resourceACL, String resourceOwner, boolean resourceOwnerIsGroup) throws UnsupportedRightException {
-        final MailboxACLEntryKey queryKey = requestUser == null ? null : new SimpleMailboxACLEntryKey(requestUser, NameType.user, false);
+    public boolean hasRight(String requestUser, GroupMembershipResolver groupMembershipResolver, Right right, MailboxACL resourceACL, String resourceOwner, boolean resourceOwnerIsGroup) throws UnsupportedRightException {
+        final EntryKey queryKey = requestUser == null ? null : new EntryKey(requestUser, NameType.user, false);
         boolean result = false;
-        Map<MailboxACLEntryKey, MailboxACLRights> entries = resourceOwnerIsGroup ? groupGlobalACL.getEntries() : userGlobalACL.getEntries();
+        Map<EntryKey, Rfc4314Rights> entries = resourceOwnerIsGroup ? groupGlobalACL.getEntries() : userGlobalACL.getEntries();
```

Comment:
maybe that's a little hard to do in this refactoring, but can you try to keep context when using generic terms like EntryKey? In this case this means using MailboxACL.EntryKey instead of just EntryKey.